app/src/main/java/com/simplecity/amp_library/utils/PlaylistUtils.java

```diff
@@ -0,0 +1,803 @@
+package com.simplecity.amp_library.utils;
+
+import android.annotation.SuppressLint;
+import android.app.Dialog;
+import android.app.ProgressDialog;
+import android.content.ContentResolver;
+import android.content.ContentUris;
+import android.content.ContentValues;
+import android.content.Context;
+import android.content.Intent;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.WorkerThread;
+import android.support.v4.util.Pair;
+import android.text.Editable;
+import android.text.Spannable;
+import android.text.SpannableStringBuilder;
+import android.text.TextUtils;
+import android.text.TextWatcher;
+import android.text.style.StyleSpan;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.SubMenu;
+import android.view.View;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.afollestad.materialdialogs.DialogAction;
+import com.afollestad.materialdialogs.MaterialDialog;
+import com.annimon.stream.Optional;
+import com.annimon.stream.Stream;
+import com.simplecity.amp_library.R;
+import com.simplecity.amp_library.ShuttleApplication;
+import com.simplecity.amp_library.interfaces.FileType;
+import com.simplecity.amp_library.model.BaseFileObject;
+import com.simplecity.amp_library.model.Playlist;
+import com.simplecity.amp_library.model.Query;
+import com.simplecity.amp_library.model.Song;
+import com.simplecity.amp_library.playback.MediaManager;
+import com.simplecity.amp_library.rx.UnsafeAction;
+import com.simplecity.amp_library.rx.UnsafeConsumer;
+import com.simplecity.amp_library.sql.SqlUtils;
+import com.simplecity.amp_library.sql.providers.PlayCountTable;
+import com.simplecity.amp_library.sql.sqlbrite.SqlBriteUtils;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
```

Comment:
Same - class needs license header, link to original source, and comment description of class.